src/Model/GiftCardChannelConfiguration.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Model;
+
+use Sylius\Component\Channel\Model\ChannelInterface as SyliusChannelInterface;
+use Sylius\Component\Locale\Model\LocaleInterface;
+
+class GiftCardChannelConfiguration implements GiftCardChannelConfigurationInterface
+{
+    /** @var int|null */
+    protected $id;
+
+    /** @var SyliusChannelInterface|null */
+    protected $channel;
+
+    /** @var LocaleInterface|null */
+    protected $locale;
+
+    /** @var GiftCardConfigurationInterface|null */
+    protected $configuration;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getChannel(): ?SyliusChannelInterface
+    {
+        return $this->channel;
+    }
+
+    public function setChannel(?SyliusChannelInterface $channel): void
```

Comment:
```suggestion
    public function setChannel(SyliusChannelInterface $channel): void
```

Reply:
This cannot be done since SyliusInterface does not allow it

Reply:
Oh yeah. Assert::notNull then I guess